### PR TITLE
fix: eagerly update experiments on successful write actions [DET-3263]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

we locally update the experiment states on successful write actions to avoid waiting for the next poll period.

depends on #640 

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/components/TaskActionDropdown.tsx [hunk 1]
```diff
@@ -53,6 +53,9 @@ const TaskActionDropdown: React.FC<Props> = ({ task }: Props) => {
       switch (params.key) { // Cases should match menu items.
         case 'kill':
           await killTask(task);
+          if (task.type === TaskType.Experiment) {
+            await updateExperimentLocally(exp => ({ ...exp, state: RunState.StoppingCanceled }));
+          } // we dont provide the immediate update for command types yet.
```
Comment (hkang1) [inline webui/react/src/components/TaskActionDropdown.tsx:58]: nit thoughts: How about conforming comments to full sentences, so starting with a capitalized letter and ending with a punctuation (which you have)? The explanation comment is great, it gives us a reason why code is "missing".  How about placing the comment right after the `if` block on a separate line so it's not on the same line as the control structure?
micro nit: `don't`

File: webui/react/src/components/TaskActionDropdown.tsx [hunk 2]
```diff
@@ -53,6 +53,9 @@ const TaskActionDropdown: React.FC<Props> = ({ task }: Props) => {
       switch (params.key) { // Cases should match menu items.
```
Comment (hkang1) [inline webui/react/src/components/TaskActionDropdown.tsx:53]: nit: Not part of this PR, but up do you. Move `// Cases should match menu items.` to the line above?

File: webui/react/src/hooks/useRestApi.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ type State<T> = {
 
 export type RestApiState<T> = State<T>;
 
-type Action<T> =
+export type Action<T> =
```
Comment (hkang1) [inline webui/react/src/hooks/useRestApi.ts:27]: question: is this necessary?

File: webui/react/src/components/TaskActionDropdown.tsx (hunk 2 above)
Comment (hamidzr) [reply webui/react/src/components/TaskActionDropdown.tsx:53]: I prefer it here but the line above works too.
I move this section in #652 or and they'll collide if I change so slightly rather not change it here
